client/consensus/src/lib.rs

```diff
@@ -149,6 +154,22 @@ impl<B, I, C> BlockImport<B> for FrontierBlockImport<B, I, C> where
 					}
 				},
 			}
+			// On importing block 1 we also map the genesis block in the auxiliary.
+			if block.header.number().clone() == One::one() {
+				let id = BlockId::Number(Zero::zero());
```

Comment:
If there is a fork at height one, this could be triggered multiple times. I'm not too worried about the performance impact, but would we be overwriting aux data in that case?

Basically, is this logic sufficiently fork-friendly?

Reply:
The auxiliary for blocks still one to many, so it will not overwrite.

When we switch to one-to-one (the switch is possible as for today, but requires migration strategy), each substrate block have a different calculated ethereum hash, so I'd say yes it's fork-friendly.